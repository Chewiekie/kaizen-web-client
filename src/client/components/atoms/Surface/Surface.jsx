import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import './Surface.scss';

const Surface = (props) => {
  const {
    children,
    className,
    disableSpacing,
    disableShadow,
  } = props;

  const surfaceClassName = clsx({
    'surface__container': true,
    '--spaced': !disableSpacing,
    '--shadowed': !disableShadow,
    [className]: className,
  });

  return (
    <div className={surfaceClassName}>
      {children}
    </div>
  );
};

Surface.propTypes = {
  /** The children to be inside the surface */
  children: PropTypes.node.isRequired,
  /** It can receive a classname to overwrites it styles */
  className: PropTypes.string,
  /** Used to remove the default padding and margin */
  disableSpacing: PropTypes.bool,
  /** Used to remove surface shadow */
  disableShadow: PropTypes.bool,
};

Surface.defaultProps = {
  className: '',
  disableSpacing: false,
  disableShadow: false,
};

export default Surface;
