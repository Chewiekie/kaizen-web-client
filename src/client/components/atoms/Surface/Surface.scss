.surface__container {
  display: inline-block;
  border-radius: getSpacingsSSK('x2');
  box-shadow: 0px 0px 4px $sskColorBoxShadowLight;

  &--no-shadow {
    box-shadow: none;
  }
}