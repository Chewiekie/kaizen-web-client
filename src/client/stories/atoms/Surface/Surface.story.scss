.surface__container--story {
    display: flex;
    flex-direction: column;
    background-color: orange;
    width: 30%;
    margin: 0 auto;
    color: white;
    padding: 15px;
    border-radius: 8px;
    & label {
        margin: 0;
        padding: 0;
        font-weight: 700;
        text-transform: uppercase;
    }
    & input {
        padding: 10px 0;
        margin: 10px 0;
    }
}