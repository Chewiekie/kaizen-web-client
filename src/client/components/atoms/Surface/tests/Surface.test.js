import React from 'react';
import { create } from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Surface from '../Surface';

configure({ adapter: new Adapter() });

describe('Surface', () => {
  test('Match Snapshot', () => {
    const surface = create(<Surface>Snapshot</Surface>);
    expect(surface.toJSON()).toMatchSnapshot();
  });

  test('Surface have class .surface__container', () => {
    const surface = shallow(<Surface>Surface</Surface>);
    expect(surface.hasClass('surface__container')).toBe(true);
  });

  test('Surface prop className', () => {
    const surface = shallow(<Surface className='surface--test'>Surface</Surface>);
    expect(surface.hasClass('surface--test')).toBe(true);
  });

  test('Surface prop children', () => {
    const surface = shallow(<Surface>Surface</Surface>);
    expect(surface.props().children).toBe('Surface');
  });

});
