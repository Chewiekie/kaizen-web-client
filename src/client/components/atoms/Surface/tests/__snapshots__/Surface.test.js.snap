// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surface Match Snapshot 1`] = `
<div
  className="surface__container "
>
  Snapshot
</div>
`;
